src/richie/apps/core/cache.py

```diff
@@ -0,0 +1,140 @@
+"""
+Plugin for django-redis that supports redis sentinel.
+Credits :
+ - https://github.com/lamoda/django-sentinel
+ - https://github.com/KabbageInc/django-redis-sentinel
+"""
+
+import logging
+import random
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+from django_redis.client import DefaultClient
+from redis.sentinel import Sentinel
+
+DJANGO_REDIS_LOGGER = getattr(settings, "DJANGO_REDIS_LOGGER", False)
+
+
+class SentinelClient(DefaultClient):
+    """
+    Sentinel client object extending django-redis DefaultClient
+    """
+
+    def __init__(self, server, params, backend):
+        """
+        Slightly different logic than connection to multiple Redis servers.
+        Reserve only one write and read descriptors, as they will be closed on exit anyway.
+        """
+        super(SentinelClient, self).__init__(server, params, backend)
+        self._client_write = None
+        self._client_read = None
+        (
+            self._master_name,
+            self._sentinel_hosts,
+            self._database_name,
+        ) = self.parse_connection_string(server)
+        self.log = logging.getLogger((DJANGO_REDIS_LOGGER or __name__))
+
+    @staticmethod
+    def parse_connection_string(constring):
+        """
+        Parse connection string in format:
+            master_name/sentinel_server:port,sentinel_server:port/db_id
+        Returns master name, list of tuples with pair (host, port) and db_id
+        """
+        try:
+            connection_params = constring.split("/")
+            master_name = connection_params[0]
+            servers = [
+                host_port.split(":") for host_port in connection_params[1].split(",")
+            ]
+            sentinel_hosts = [(host, int(port)) for host, port in servers]
+            database_name = connection_params[2]
+        except (ValueError, TypeError, IndexError):
+            raise ImproperlyConfigured("Incorrect format '%s'" % (constring))
+
+        return master_name, sentinel_hosts, database_name
+
+    def get_client(self, write=True, tried=(), show_index=False):
+        """
+        Method used to obtain a raw redis client.
+
+        This function is used by almost all cache backend
+        operations to obtain a native redis client/connection
+        instance.
+        """
+        self.log.debug("get_client called: write=%s", write)
+        if write:
+            if self._client_write is None:
+                self._client_write = self.connect(write)
+
+            if show_index:
+                return self._client_write, 0
+            return self._client_write
```

Comment:
hmm returning sometimes a tuple, sometimes just the connection. Maybe we should improve typing by returning always the same thing?

Reply:
This behavior is required by the `django_redis` library.
See [here](https://github.com/jazzband/django-redis/blob/879cbc270466a2d731f369e7cfede375705a0fc5/django_redis/client/default.py#L82-L98).

I'm not comfortable with changing the method prototype without patching it in the `django_redis` library.